### PR TITLE
5648 isapi lookup v2

### PR DESCRIPTION
**Pull request by alibombaye**
a cleaner solution

### CODE REVIEW

File: app/server.js [hunk 1]
```diff
@@ -1,11 +1,42 @@
...
 const { logger } = require('./logger');
 
+const setTimeoutPromise = util.promisify(setTimeout);
+
+const determineAppTypeToStart = async ({
+  canStartApp, appType, attempt, pollDuration,
+}) => {
+  if (!canStartApp) {
+    if (config.loginEnabled === 'false') {
+      return determineAppTypeToStart({
```
Comment (kat1906) [inline app/server.js:17]: arggghhhhh recursion 😭

File: app/server.js [hunk 2]
```diff
@@ -15,44 +46,25 @@ const { logger } = require('./logger');
     }
   });
 
-  let intervalId;
-  let app;
-  const pollDuration = 1000;
+  const appType = await determineAppTypeToStart({
+    canStartApp: false, appType: undefined, attempt: 1, pollDuration: 1000,
```
Comment (kat1906) [inline app/server.js:50]: you don't need `appType: undefined` as it will be undefined anyway

File: app/server.js (hunk 1 above)
Comment (alibombaye) [reply app/server.js:17]: I dreamt this solution last night and was so excited to try it out this morning 😂

File: app/server.js (hunk 2 above)
Comment (alibombaye) [reply app/server.js:50]: gives me a lint error when I don't pass anything in

Comment (kat1906) [reply app/server.js:50]: put default into the function?

Comment (kat1906) [reply app/server.js:50]: I don't see why you can't set the appType before it goes into the function?

Comment (alibombaye) [reply app/server.js:50]: still doesn't like that. Is it perhaps due to it being an object that is passed in

File: app/server.js [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
 
+const setTimeoutPromise = util.promisify(setTimeout);
+
+const isIsapiReady = async ({
+  isapiReady, attempt, pollDuration,
+}) => {
+  if (!isapiReady) {
+    try {
+      await axios.get(`${config.oidcBaseUri}/.well-known/openid-configuration`);
+      return isIsapiReady({
```
Comment (kat1906) [inline app/server.js:18]: can you just return true here?

Comment (alibombaye) [reply app/server.js:18]: Yes I can 😄 got a little recursive mad

File: app/server.js [hunk 4]
```diff
@@ -1,11 +1,34 @@
...
+      const nextAttempt = attempt + 1;
+      const nextPollDuration = nextAttempt * pollDuration;
+      logger.error(`Isapi is not ready - will poll again in ${nextAttempt} seconds`);
+      return setTimeoutPromise(nextPollDuration).then(() => isIsapiReady({
+        isapiReady: false, attempt: nextAttempt, pollDuration,
+      }));
+    }
+  }
+
+  return isapiReady;
```
Comment (kat1906) [inline app/server.js:29]: I think this is redundant now

Comment (kat1906) [reply app/server.js:29]: actually the whole if statement

Comment (alibombaye) [reply app/server.js:29]: I'll get a lint error if I remove that

File: app/server.js [hunk 5]
```diff
@@ -15,44 +38,31 @@ const { logger } = require('./logger');
...
-  let app;
-  const pollDuration = 1000;
+  let appType;
+  if (config.loginEnabled === 'false') {
+    appType = 'basic';
+  } else {
+    await isIsapiReady({
+      isapiReady: false, attempt: 1, pollDuration: 1000,
+    });
+    appType = 'auth';
```
Comment (kat1906) [inline app/server.js:48]: if you return `'auth'` from the `isIsapiReady` function then you could do something like this:
`const appType = (config.loginEnabled === 'false') ?  'basic' : await isIsapiReady({ attempt: 1, pollDuration: 1000 });`

File: app/server.js (hunk 4 above)
Comment (alibombaye) [reply app/server.js]: 😍

File: app/server.js [hunk 6]
```diff
@@ -1,11 +1,30 @@
...
 const { routes } = require('./routes');
 const { logger } = require('./logger');
 
+const setTimeoutPromise = util.promisify(setTimeout);
+
+const isIsapiReady = async ({
+  attempt, pollDuration,
+}) => {
+  try {
+    await axios.get(`${config.oidcBaseUri}/.well-known/openid-configuration`);
```
Comment (kat1906) [inline app/server.js:16]: maybe a logger comment here too?

File: app/server.js (hunk 5 above)
Comment (alibombaye) [reply app/server.js]: Done similar to what you suggested. I like isIsapiReady returning just a boolean

**Comment by kat1906** [comment]
This looks FAR better than the original, bravo 👏

**Pull request merged**